### PR TITLE
feat(cli): create plugin for handling application api calls

### PR DESCRIPTION
**Pull request by odinr**
## Summary
This pull request introduces the `appProxyPlugin`, a Vite plugin designed to streamline the development process for Fusion apps by proxying API and bundle requests to the Fusion app backend.

## Why This Plugin is Required
The `appProxyPlugin` addresses several key challenges faced during the development of Fusion apps:

1. **Token Preservation**: Ensures authorization tokens are preserved and reused for subsequent requests, simplifying authentication during development.
2. **Proxy Configuration**: Sets up proxy rules for API and bundle requests, allowing developers to work with local and remote resources seamlessly.
3. **App Configuration and Manifest**: Serves the app configuration and manifest based on the app key and version, ensuring the correct assets and settings are used during development.

## Example Usage
```typescript
const viteConfig = defineConfig({
    // vite configuration
    plugins: [
        appProxyPlugin({
            proxy: {
                path: '/app-proxy',
                target: 'https://fusion-s-apps-ci.azurewebsites.net/',
                // optional callback when matched request is proxied
                onProxyReq: (proxyReq, req, res) => {
                    proxyReq.on('response', (res) => { console.log(res.statusCode) });
                },
            },
            // optional, but required for serving local app configuration, manifest and resources
            app: {
                key: 'my-app',
                version: '1.0.0',
                generateConfig: async () => ({}),
                generateManifest: async () => ({}),
            },
        }),
    ],
});

// Example API calls
fetch('/app-proxy/apps/my-app/builds/1.0.0/config'); // local
fetch('/app-proxy/apps/my-app/builds/0.0.9/config'); // proxy
fetch('/app-proxy/apps/other-app/builds/1.0.0/config'); // proxy

// Example asset calls
fetch('/app-proxy/bundles/my-app/builds/1.0.0/index.js'); // local
fetch('/app-proxy/bundles/my-app/builds/0.0.9/index.js'); // proxy
```

### Check off the following:
- [x] Confirm that I checked changes to branch which I am merging into.
  - _I have validated included files_
  - _My code does not generate new linting warnings_
  - _My PR is not a duplicate, [check existing pr`s](https://github.com/equinor/fusion-framework/pulls)_
- [x] Confirm that the I have completed the [self-review checklist](https://github.com/equinor/fusion-framework/blob/main/contributing/self-review.md).

- [x] Confirm that my changes meet our [code of conduct](https://github.com/equinor/fusion-framework/blob/main/CODE_OF_CONDUCT.md).

### CODE REVIEW

File: packages/cli/src/bin/plugins/app-proxy.ts [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+        apply: 'serve',
+        config(config) {
+            config.server ??= {};
+            config.server.proxy = {
+                // proxy all api calls to the fusion apps backend
+                [`${proxyPath}/apps`]: {
+                    target: 'https://fusion-s-apps-ci.azurewebsites.net/',
+                    changeOrigin: true,
+                    secure: false,
+                    rewrite: (path) => path.replace(/^\/app-proxy/, ''),
```
Comment (Noggling) [inline packages/cli/src/bin/plugins/app-proxy.ts:86]: Should the replace here use the proxyPath? sins this is an input to the plugin?

File: packages/cli/src/bin/plugins/app-proxy.ts [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
+                        });
+                        proxy.on('proxyReq', proxyRequestLogger);
+                    },
+                },
+                // proxy all bundle requests to the fusion apps backend
+                [`${proxyPath}/bundles`]: {
+                    target: 'https://fusion-s-apps-ci.azurewebsites.net/',
+                    changeOrigin: true,
+                    secure: false,
+                    rewrite: (path) => path.replace(/^\/app-proxy/, ''),
```
Comment (Noggling) [inline packages/cli/src/bin/plugins/app-proxy.ts:102]: Same as over?

File: packages/cli/src/bin/plugins/app-proxy.ts (hunk 1 above)
Comment (odinr) [reply packages/cli/src/bin/plugins/app-proxy.ts:86]: yupp, had it in the back of my mind :) but to trigger happy

**Review by Noggling — APPROVED** [review]
LGTM

**Pull request merged**